tests/functional/backend/corpora/test_api.py

```diff
@@ -199,23 +199,24 @@ def test_dataset_upload_flow(self):
 
         with self.subTest("Test dataset conversion"):
             keep_trying = True
-            upload_statuses = ["WAITING", "UPLOADING", "UPLOADED"]
-            conversion_statuses = ["CONVERTING", "CONVERTED", "FAILED"]
+            expected_upload_statuses = ["WAITING", "UPLOADING", "UPLOADED"]
+            # conversion statuses can be `None` when/if we hit the status endpoint too early after an upload
+            expected_conversion_statuses = ["CONVERTING", "CONVERTED", "FAILED", None]
```

Comment:
Since this test hits the `status` endpoint immediately after `upload-links`, sometimes it was hitting it before the conversion began and no conversion statuses exist yet. In this case, `None` would be the expected conversion status.